media-gfx/sane-backends/sane-backends-1.4.0.ebuild

```diff
@@ -0,0 +1,368 @@
+	umax_pp
+	xerox_mfp
+)
+
+IUSE="gphoto2 +jpeg png snmp systemd threads +tiff +usb v4l xinetd +zeroconf"
+
+for GBACKEND in ${IUSE_SANE_BACKENDS[@]}; do
+	case ${GBACKEND} in
+	# Disable backends that require parallel ports as no one has those anymore.
+	canon_pp|hpsj5s|mustek_pp|\
+	pnm|mustek_usb2|kvs40xx)
+		IUSE+=" sane_backends_${GBACKEND}"
+		;;
+	*)
+		IUSE+=" +sane_backends_${GBACKEND}"
+	esac
+done
+
+REQUIRED_USE="
+	gphoto2? ( jpeg )
+	sane_backends_dc210? ( jpeg )
+	sane_backends_dc240? ( jpeg )
+	sane_backends_dell1600n_net? ( jpeg tiff )
+	sane_backends_epsonds? ( jpeg )
+	sane_backends_escl? ( jpeg zeroconf )
+	sane_backends_kvs40xx? ( threads )
+	sane_backends_mustek_usb2? ( threads )
+	sane_backends_pixma? ( jpeg )
+"
+
+# Build system uses automagic to find libraries
+# Some of these are then unconditionally required for some drivers.
+# See acinclude.m4 for details
+# libxml2: required for escl and --with-usb-record-replay (could be made optional)
+RDEPEND="
+	acct-user/saned
+	acct-group/scanner
+	dev-libs/libxml2:=[${MULTILIB_USEDEP}]
+	gphoto2? ( >=media-libs/libgphoto2-2.5.3.1:=[${MULTILIB_USEDEP}] )
+	jpeg? ( media-libs/libjpeg-turbo:=[${MULTILIB_USEDEP}] )
+	png? ( media-libs/libpng:=[${MULTILIB_USEDEP}] )
+	sane_backends_canon_pp? ( >=sys-libs/libieee1284-0.2.11-r3[${MULTILIB_USEDEP}] )
+	sane_backends_escl? (
+		dev-libs/glib:2
+		app-text/poppler[cairo]
+		|| (
+			net-dns/avahi[dbus]
+			net-dns/avahi[gtk]
+		)
+		net-dns/avahi
+		net-misc/curl
+		x11-libs/cairo
+	)
+	sane_backends_hpsj5s? ( >=sys-libs/libieee1284-0.2.11-r3[${MULTILIB_USEDEP}] )
+	sane_backends_mustek_pp? ( >=sys-libs/libieee1284-0.2.11-r3[${MULTILIB_USEDEP}] )
+	snmp? ( net-analyzer/net-snmp:= )
+	systemd? ( sys-apps/systemd:= )
+	tiff? ( >=media-libs/tiff-3.9.7-r1:=[${MULTILIB_USEDEP}] )
+	usb? ( >=virtual/libusb-1-r1:1[${MULTILIB_USEDEP}] )
+	v4l? ( >=media-libs/libv4l-0.9.5[${MULTILIB_USEDEP}] )
+	xinetd? ( sys-apps/xinetd )
+	zeroconf? ( >=net-dns/avahi-0.6.31-r2[${MULTILIB_USEDEP}] )
+"
+DEPEND="${RDEPEND}
+	v4l? ( sys-kernel/linux-headers )
+"
+BDEPEND="
+	dev-build/autoconf-archive
+	sys-devel/gettext
+	virtual/pkgconfig
+	sane_backends_pixma? ( ${PYTHON_DEPS} )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-1.0.30-add_hpaio_epkowa_dll.conf.patch
+	"${FILESDIR}"/${PN}-1.4.0-remove-automagic.patch
+)
+
+# ./configure checks to see if these are defined in sys/io.h,
+# so if they're undefined... that's the point.
+QA_CONFIG_IMPL_DECL_SKIP=( inb outb )
+
+MULTILIB_CHOST_TOOLS=(
+	/usr/bin/sane-config
+)
+
+src_prepare() {
+	default
+
+	echo "${PV}" > .tarball-version || die
+	eautoreconf
+
+	# Needed for udev rules generation/installation
+	multilib_copy_sources
+}
+
+src_configure() {
+	# From Fedora
+	append-flags -fno-strict-aliasing
+
+	multilib-minimal_src_configure
+}
+
+multilib_src_configure() {
+	# the blank is intended - an empty string would result in building ALL backends.
+	local lbackends=" "
+
+	use gphoto2 && lbackends="gphoto2"
```

Comment:
pkg-config is hardcoded for gphoto2 in acinclude.m4
It will require at least s/pkg-config/$PKG_CONFIG/

Reply:
> It will require at least s/pkg-config/$PKG_CONFIG/

... and export PKG_CONFIG in the ebuild

And upstreamable patch would use autotools functions just like they are using in configure.ac for systemd and friends.